gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/UserTaskController.java

```diff
@@ -54,6 +60,34 @@ public CompletableFuture<ResponseEntity<Object>> completeUserTask(
                 .orElseGet(() -> ResponseEntity.noContent().build()));
   }
 
+  @PostMapping(
+      path = "/user-tasks/{userTaskKey}/assignment",
+      produces = "application/json",
+      consumes = "application/json")
+  public CompletableFuture<ResponseEntity<Object>> assignUserTask(
+      final ServerWebExchange context,
+      @PathVariable final long userTaskKey,
+      @RequestBody() final UserTaskAssignmentRequest assignmentRequest) {
+
+    final Optional<ResponseEntity<Object>> validationResult =
+        validateAssignmentRequest(assignmentRequest);
+
+    if (validationResult.isPresent()) {
+      return CompletableFuture.completedFuture(validationResult.get());
+    }
```

Comment:
🔧 Can we use the result name to indicate what we receive here and why we want to return it? We're optionally receiving a `validationErrorResponse`, I think. If it's present, we want to return it. Otherwise, reading it, one could wonder why we return a validation result. Is that a positive result? Or a negative one? Or do we return in both cases?

```suggestion
    final Optional<ResponseEntity<Object>> validationErrorResponse =
        validateAssignmentRequest(assignmentRequest);

    if (validationErrorResponse.isPresent()) {
      return CompletableFuture.completedFuture(validationErrorResponse.get());
    }
```

Reply:
👍  I agree `validationErrorResponse` makes the code more clear and readable